packages/api/src/utils/billing-types.ts

```diff
@@ -0,0 +1,66 @@
+export type StripePaymentMethodId = string;
+export type CustomerId = string;
+
+export type StripeCardPaymentMethod = {
+  id: `pm_${string}`
+  card: {
+    '@type': 'https://stripe.com/docs/api/cards/object'
+    brand: string
+    country: string | null
+    exp_month: number
+    exp_year: number
+    funding: string
+    last4: string
+  }
+}
+
+export type IdOnlyPaymentMethod = {
+  id: string
+}
+
+export type PaymentMethod = StripeCardPaymentMethod | IdOnlyPaymentMethod
+
+/**
+ * Indicates that a process was not able to find a specific Customer record.
+ */
+export interface CustomerNotFound extends Error {
+  code: 'ERROR_CUSTOMER_NOT_FOUND'
+}
+
+export interface BillingService {
+  getPaymentMethod(customer: CustomerId): Promise<PaymentMethod|null|CustomerNotFound>
+  savePaymentMethod(
+    customer: CustomerId,
+    paymentMethodId: StripePaymentMethodId
+  ): Promise<void>;
+}
+
+export interface Customer {
+  id: string
+}
+
+export interface CustomersService {
```

Comment:
I really like the distinct types for `Customer` vs `User`. It seems like the kind of thing that often gets bundled together needlessly.

Reply:
glad it seems sensible. the way things are configured in prod, there's definitely an important distinction because the system of record for Users is in the web3.storage db, but the system of record for the (sometimes) associated customer is stripe.com.